File: lifecycle-api/models/api/webhooks/connection.py
from typing import Optional
from pydantic import BaseModel
from models.enums.rfc23 import ConnectionState as ConnectionStateRFC
from models.enums.connection import State

class WebhookConnectionData(BaseModel):
    state: State
    created_at: str
    updated_at: str
    connection_id: str
    their_did: Optional[str] = None
    their_label: Optional[str] = None
    their_role: str
    connection_protocol: str
    rfc23_state: ConnectionStateRFC
    invitation_key: str
    accept: str
    invitation_mode: str
    alias: str

File: controller/src/app/models/connection.ts
import { AcceptType, ConnectionStatus, InvitationMode, RoutingStatus } from "../enums/connection-status";

export type Connection = {
  initiator: string;
  invitation_key: string;
  routing_state: RoutingStatus;
  my_did: string;
  state: ConnectionStatus;
  connection_id: string;
  request_id: string;
  their_role: string;
  their_label: string;
  alias: string;
  error_msg: string;
  inbound_connection_id: string;
  accept: AcceptType;
  invitation_mode: InvitationMode;
  their_did: string;
  created_at: Date;
  updated_at: Date;
};


